be/src/storage/delta_writer.h

```diff
@@ -64,6 +64,7 @@ struct DeltaWriterOptions {
     ReplicaState replica_state;
     bool miss_auto_increment_column = false;
     bool abort_delete = false;
+    PartialUpdateMode partial_update_mode = PartialUpdateMode::ROW_MODE;
```

Comment:
1. It's better to determine the ROW_MODE or COLUMN_MODE when writing.
2. If using the `partial_update_mode` in the stream load, the user can not get an idea of how many segments will be included in one ingestion.

Reply:
After discussion, keep `partial_update_mode`, and add `PartialUpdateMode::AUTO_MODE`, make it a session variable. Using `partial_update_mode=PartialUpdateMode::AUTO_MODE` as default value, SR will decide  ROW_MODE or COLUMN_MODE when writing.